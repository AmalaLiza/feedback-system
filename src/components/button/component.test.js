import React from "react";
import { shallow } from "enzyme";
import Button from "./component";
import renderer from "react-test-renderer";

const props = {
	buttonText: "Button title",
	onClick: (noop) => noop,
};

function setup() {
	const wrapper = shallow(<Button {...props} />);
	return { wrapper, props };
}

describe("Button Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<Button {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should have a button", () => {
		const { wrapper } = setup();
		expect(wrapper.find("button").exists()).toBe(true);
	});
});
