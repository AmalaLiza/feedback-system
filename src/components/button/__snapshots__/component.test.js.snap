// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Test Suite renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Button title
</button>
`;
